plugins/rest/rest_auth.go

```diff
@@ -113,33 +138,116 @@ type tokenEndpointResponse struct {
 // oauth2ClientCredentialsAuthPlugin represents authentication via a bearer token in the HTTP Authorization header
 // obtained through the OAuth2 client credentials flow
 type oauth2ClientCredentialsAuthPlugin struct {
-	TokenURL     string    `json:"token_url"`
-	ClientID     string    `json:"client_id"`
-	ClientSecret string    `json:"client_secret"`
-	Scopes       *[]string `json:"scopes,omitempty"`
-
-	tokenCache    *oauth2Token
-	tlsSkipVerify bool
+	GrantType    string                 `json:"grant_type"`
+	TokenURL     string                 `json:"token_url"`
+	ClientID     string                 `json:"client_id"`
+	ClientSecret string                 `json:"client_secret"`
+	SigningKeyID string                 `json:"signing_key"`
+	Claims       map[string]interface{} `json:"additional_claims"`
+	IncludeJti   bool                   `json:"include_jti_claim"`
+	Scopes       *[]string              `json:"scopes,omitempty"`
+
+	signingKey       *keys.KeyConfig
+	signingKeyParsed interface{}
+	tokenCache       *oauth2Token
+	tlsSkipVerify    bool
 }
 
 type oauth2Token struct {
 	Token     string
 	ExpiresAt time.Time
 }
 
+func (ap *oauth2ClientCredentialsAuthPlugin) createAuthJwt(claims map[string]interface{}, signingKey interface{}) (*string, error) {
+	now := time.Now()
+	baseClaims := map[string]interface{}{
+		"iat": now.Unix(),
+		"exp": now.Add(1 * time.Hour).Unix(),
```

Comment:
💭 Would anyone want to configure this?

Reply:
I doubt it - what we issue here is essentially a single use (i.e. one time) token. The access token we get in the response is the one that will be cached and reused for its lifetime, and as soon as that one is expired we craft a new JWT.. so the expiry should not matter here, unless you're on a _really_ slow connection ;)

Reply:
Sounds like we could go with something like 3 or 5 or 10 minutes then...? 😉

Reply:
Yes, usually its 5-10 minutes. Less is better.

Reply:
The use case we catered to here was primarily GCP, where an hour is the max. I don't mind lowering this to 10 minutes, so I'll do that.